src/applications/financial-status-report/actions/geographicMeansThreshold.js

```diff
@@ -1,48 +1,107 @@
+  const discretionaryIncomeThreshold =
+    gmtValue * DISCRETIONARY_INCOME_PERCENTAGE;
+
+  return {
+    incomeUpperThreshold,
+    assetThreshold,
+    discretionaryIncomeThreshold,
+  };
+};
+
+/**
+ * Determines the appropriate data source based on configuration settings and environment.
+ *
+ * @returns {string} The URL of the data source.
+ */
+
+const getDataUrl = (dependents, year, zipCode) => {
+  const isLocalhost = environment.isLocalhost();
+  if (isLocalhost && USE_GEOGRAPHIC_MOCK_DATA) {
+    return null; // Return mock data directly
+  }
 
-const INCOME_UPPER_PERCENTAGE = 1.5;
-const ASSET_PERCENTAGE = 0.065;
-const DISCRETIONARY_INCOME_PERCENTAGE = 0.0125;
+  if (isLocalhost && USE_STAGING_DATA) {
+    return `https://staging-api.va.gov/income_limits/v1/limitsByZipCode/${zipCode}/${year}/${dependents}`;
+  }
 
-export const getGMT = (dependents, year, zipCode) => {
-  const CONTEXT_ROOT = '/income_limits/v1/limitsByZipCode';
-  const REQUEST_URL = `${
+  return `${
     environment.API_URL
-  }${CONTEXT_ROOT}/${zipCode}/${year}/${dependents}`;
-
-  // For testing locally if api db isn't populated
-  // const REQUEST_URL = `https://staging-api.va.gov/income_limits/v1/limitsByZipCode/${zipCode}/${year}/${dependents}`;
-
-  return apiRequest(REQUEST_URL)
-    .then(({ data }) => {
-      // incomeUpperThreshold is 150% of the GMT
-      const incomeUpperThreshold = data.gmtThreshold * INCOME_UPPER_PERCENTAGE;
-      // assetThreshold is 6.5% of the GMT
-      const assetThreshold = data.gmtThreshold * ASSET_PERCENTAGE;
-      // discretionaryStatus is 1.25% of the GMT
-      const discretionaryIncomeThreshold =
-        data.gmtThreshold * DISCRETIONARY_INCOME_PERCENTAGE;
-      const gmtThresholdError =
-        typeof data.gmtThreshold === 'number'
-          ? null
-          : 'GMT threshold is not a number';
-
-      return {
-        ...data,
-        error: gmtThresholdError,
-        incomeUpperThreshold,
-        assetThreshold,
-        discretionaryIncomeThreshold,
-      };
-    })
-    .catch(({ error }) => {
-      Sentry.withScope(scope => {
-        scope.setExtra('error', error);
-        Sentry.captureMessage(`income_limits failed: ${error}`);
-      });
-
-      return { gmtThreshold: null, error };
+  }/income_limits/v1/limitsByZipCode/${zipCode}/${year}/${dependents}`;
+};
+
+/**
+ * Retrieves GMT thresholds data. Depending on configuration, it either fetches data from an API,
+ * uses staging data, or returns mock data (if enabled).
+ *
+ * @param {number} dependents - The number of dependents.
+ * @param {number} year - The year for which data is being requested.
+ * @param {string} zipCode - The ZIP code for which data is being requested.
+ * @returns {Promise<Object>} A promise resolving to the GMT thresholds data or error object.
+ */
+export const getGMT = async (dependents, year, zipCode) => {
+  const dataUrl = getDataUrl(dependents, year, zipCode);
+
+  if (dataUrl === null) {
```

Comment:
I know we talked about this already, but the `null` just gives me the heebie jeebies. Still going to slap an approval on it, but a soft suggestion of maybe setting `dataUrl` to a string like 'useMock' and have the check there

Reply:
Added  a check of isLocalhost for an extra layer of protection 👍